week2ShortAnswers

```diff
@@ -0,0 +1,100 @@
+
+-Continuous Learning is the idea that you continue learning new ideas and new techniques throughout your career to stay relevant and to keep your mind active longer. 
+Continuous learning involves focusing on a subject and setting up a learning path to learn new knowledge while also building and keeping what you already know fresh. 
+
+c) How can you find out what modifiers a View has?
+
+You can look in the library or you can put a period “.” And a list will populate
+
+d) What is a breakpoint?
+
+A breakpoint is like a stop sign. You place a breakpoint on a line and it enables you to view the state of the variables in that given line of code. Used in debugging code when you receive an unexpected result. 
+
+e) How can you access environment values in your App?
+
+You use @Enviroment
+@Enviroment(\.keypath) var propertName
+
+f) How can you determine, in code, if the App is in Dark or Light Mode?
+
+You can use the environment property 
+@Enviroment(\.colorScheme) var colorScheme
+
+g) Why are magic numbers an issue, and how should you avoid them?
+
+Magic numbers are numbers that are hard-coded into your code. They should be avoided because it’s hard to tell where they come from. Turning them into constants avoids that confusion.  
+
+h) How can you view your App in Light and Dark Modes simultaneously?
+
+You can manually add another view by copying the current instance of the preview and you can change and modify the view by choosing another device, orientation, and dark/regular view. 
+
+i) Below is an image of the Canvas from Xcode. The Canvas is in selectable mode. Can you explain why the red background does not cover the entire button area?
+￼
+
+Because a view only takes up as much as space as needed. With this error you would want to ensure to implement the frame before the background.
+
+j) Modifier padding(10) adds padding to the view's top, bottom, left, and right sides. How could a padding of 10 be added to only the left and right sides of the view? The answer for this question should be a short section of code.
+
+.padding(.horizontal)
+
+k) Provide two reasons why you would want to extract views.
+
+One of the main reasons to extract code is to keep the code organized, modular, and reusable. 
+This makes it easier to make changes in the app. You only need to go to one view to make the changes
+
+l) How can you determine, in code, if the device is in Portrait or Landscape mode?
+
+@Environment(\.verticalSizeClass) var verticalSizeClass
+@Environment(\.horizontalSizeClass) var horizantalSizeClass
+
+If verticalSizeClass == .regular &&
+    horizontalSizeClass == .compact {
+Code
+}
+
+m) What is a literal value?
+
+A literal value is any number without a decimal point. 
+
+o) What are the safe areas?
+
+It’s the top of the screen with the battery indicator. Not somewhere where you would want to put any writing. But if you want to ignore the safe area you can use “.ignoreSafeArea()”
```

Comment:
It's not just the top of the screen. Safe areas are guidelines that iOS gives you depending on the device. So, for example, you have the top safe area, that prevents you from putting views on top or behind ares of the screen that could be obstructed by the hardware or system UI, for example the battery indicator. There's also the bottom guide, that prevents you from putting views behind the "home" indicator (the indicator to swipe to go home on iPhones and iPads). You also have the trailing and leading guides, that might feel irrelevant because there's usually no UI or hardware on the right or left side of the device, but are very important when you're working with landscape orientation (because then you have the dynamic island and such).